### ISSUE TITLE
Test in Safari

### ISSUE THREAD

**Issue by benrbray** [benrbray/prosemirror-math#11]
ProseMirror (and by extension, `prosemirror-math`) relies on `contenteditable` to function.  Since the implementation of `contenteditable` is not yet standardized between browsers, we must test manually in each browser to ensure acceptable behavior.

I do not own a Mac, so I cannot test `prosemirror-math` on Safari.  Please report any bugs you find in Safari!

**Comment by benrbray** [comment]
In the meantime, it looks like I can use the webkit-based [Epiphany browser](https://wiki.gnome.org/Apps/Web) as an approximation to Safari.  However, the developer tools are lacking (even `console.log` is unsupported).

**Comment by gzuidhof** [comment]
This package currently doesn't work in Safari, I just gave it a try. A prosemirror editor won't be created at all as a regex literal is invalid:

```
SyntaxError: Invalid regular expression: invalid group specifier name
```

The issue appears to be lack of negative lookahead support (which surely is also an issue in [not-that-old browsers](https://caniuse.com/?search=lookbehind)).

Perhaps we can try-catch that part and print a warning to the console in case it is not supported? These browsers would then only be able to enter math editing mode by pressing a button or something which seems acceptable for now. A more long-term fix would be to implement [one of these](https://stackoverflow.com/questions/641407/javascript-negative-lookbehind-equivalent) approaches.

---

As for not owning a Mac, I have used Browserstack in the past for testing on a range of (very) old devices and browsers. You can get a free account through their generous free Open Source offering (https://www.browserstack.com/open-source). I got approved pretty quickly, perhaps this could be useful for your open source work :)

**Comment by benrbray** [comment]
Thanks for trying it out in Safari.  Are you using the most up-to-date version?  Is the issue present on the [demo page](https://benrbray.com/prosemirror-math/)?  Based on [some user feedback](https://physics.meta.stackexchange.com/a/13489/294610), I already [attempted to fix](https://github.com/benrbray/prosemirror-math/blob/master/src/plugins/math-inputrules.ts#L20) the unsupported regex issue, but if it's still a problem I'll take another look.

I'll also look into Browserstack, thanks for the suggestion!

**Comment by gzuidhof** [comment]
You're right, sorry about the false alarm :(. I naively opened Starboard's homepage, but I didn't actually push Starboard's latest version to there. You can close this ticket!
